docs/cli/discount-code-generator.md

```diff
@@ -0,0 +1,153 @@
+# Discount Code Generator
+
+Generate unique discount codes to be imported to the [commercetools platform](https://dev.commercetools.com/). See usage below
+
+## Usage
+`npm install @commercetools/discount-code-generator --global`
+
+### CLI
+```
+Usage: discountCodeGenerator [options]
+
+Options:
+  --help, -h                 Show help text.                                                     [boolean]
+  --version, -v              Show version number.                                                [boolean]
+  --quantity, -q             Quantity of discount codes to generate. (Between 1 and 500000)     [required]
+  --code-length, -l          Length of the discount codes to generate.                       [default: 11]
+  --code-prefix, -p          Prefix for each code. No prefix will be used if omitted.        [default: ""]
+  --input, -i                Path to code options CSV or JSON file.
+  --output, -o               Path to store generated output file.                      [default: "stdout"]
+  --delimiter, -d            Used CSV delimiter for input and/or output file.               [default: ","]
+  --multivalueDelimiter, -m  Used CSV delimiter in multivalue fields for input/output file. [default: ";"]
+```
+
+#### Info on flags
+- The ```--quantity``` flag represents the number of codes to be generated. It must be a number between 1 and 500000
+- The ```--code-length``` flag is used to customise the discount codes according to individual use cases. (more info in Exmples)
+- The ```--input``` flag specifies the path to a CSV or JSON file containing the options the discount codes should have. If this flag is omitted, discount codes without any attributes will be generated.
+- The ```--output``` flag specifies where to output the generated codes. Several notes on this flag:
+  - If the file specified already exists, it will be overwritten.
+  - The default location for status report logging is the standard output.
+  - If no output path is specified, the generated codes will be logged to the standard output as a result, status reports will be logged to a `discountCodeGenerator.log` file
+- The ```--delimiter``` flag specifies the delimiter used in the input file and/or output file if any or both are CSV. Defaults to ```','``` if omitted and will be ignored if neither input nor output is CSV
+- The ```--multivalueDelimiter``` flag specifies the delimiter for arrays in CSV. Note that only the ```cartDiscounts``` field if present should be an array. Defaults to ```';'``` if omitted
```

Comment:
There is actually no array in the CSV, it's just a delimiter for multi values in a cell... if you get what I mean...